### PR TITLE
Init "drone-cobertura-s3" plugin push

### PR DESCRIPTION
**Pull request by shiv-am0**

### CODE REVIEW

**Review by Ompragash — CHANGES_REQUESTED** [review]
All looks good except the LICENSE file. Fix it by not modifying the License file content in your Push changes and it's good to be merged.

**Comment by shiv-am0** [comment]
@Ompragash I made the the changes as asked. Kindly review it.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+FROM amazon/aws-cli:amd64
+
```
Comment (hemanthmantri) [inline Dockerfile:2]: Do we need a Dockerfile for each arch?

File: README.md [hunk 2]
```diff
@@ -1 +1,78 @@
-# drone-cobertura-s3
\ No newline at end of file
+# drone-cobertura-s3
+
+Drone plugin to upload cobertura code coverage reports to AWS S3 bucket and publish the bucket static site url to `Artifacts` tab under `Pipieline Execution`.
```
Comment (hemanthmantri) [inline README.md]: s/url/URL

File: README.md [hunk 3]
```diff
@@ -1 +1,78 @@
...
+```bash
+go build
+```
+
+## Docker
+
+Build the Docker image with the following commands:
+
+```
+./hacking/build.sh
```
Comment (hemanthmantri) [inline README.md]: Why is the directory named hacking here? Please consider using a better name.

File: README.md [hunk 4]
```diff
@@ -1 +1,78 @@
...
+   drone-cobertura-s3
+...
+...
+GLOBAL OPTIONS:
+   --pipeline-sequence-id value  Harness CIE Pipeline Sequence ID [$PLUGIN_PIPELINE_SID]
+   --aws-access-key value        AWS Access Key ID [$PLUGIN_AWS_ACCESS_KEY_ID]
+   --aws-secret-key value        AWS Secret Access Key [$PLUGIN_AWS_SECRET_ACCESS_KEY]
+   --aws-default-region value    AWS Default Region [$PLUGIN_AWS_DEFAULT_REGION]
+   --aws-bucket value            AWS Default Region [$PLUGIN_AWS_BUCKET]
+   --report-source value         AWS Default Region [$PLUGIN_REPORT_SOURCE]
```
Comment (hemanthmantri) [inline README.md]: Fix the descriptions. Looks like they are copy-paste errors.

File: README.md [hunk 5]
```diff
@@ -1 +1,78 @@
...
+In Harness CI,
+Using AWS Access Key ID and Secret Access Key as authentication method:
+```yaml
+              - step:
+                  type: Plugin
+                  name: drone_cobertura_s3
+                  identifier: drone_cobertura_s3
+                  spec:
+                    connectorRef: account.harnessImage
+                    image: harnesscommunitytest/drone-cobertura-s3
```
Comment (hemanthmantri) [inline README.md:55]: Why is this not a first class plugin in Drone plugins repository? Eg: `plugins/cobertura-s3`

File: main.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+)
+
+var (
+	pluginVersion = "0.0.1"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "drone-cobertura-s3"
+	app.Usage = "Drone plugin to upload cobertura code coverage reports to AWS S3 bucket and publish the s3 bucket static site url under 'Executions > Artifacts' tab"
```
Comment (hemanthmantri) [inline main.go]: URL

File: main.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+			EnvVar: "PLUGIN_ROLE_ARN",
+		},
+		cli.StringFlag{
+			Name:   "role-session-name",
+			Usage:  "AWS Role Session Name",
+			EnvVar: "PLUGIN_ROLE_SESSION_NAME",
+		},
+		cli.StringFlag{
+			Name:   "report-source",
+			Usage:  "AWS Default Region",
```
Comment (hemanthmantri) [inline main.go]: Fix descriptions

File: main.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+			EnvVar: "PLUGIN_REPORT_SOURCE",
+		},
+		cli.StringFlag{
+			Name:   "report-target",
+			Usage:  "Report target",
```
Comment (hemanthmantri) [inline main.go]: Not useful usage - same as name. Please add more desc and example.

File: Dockerfile (hunk 1 above)
Comment (shiv-am0) [reply Dockerfile:2]: This plugin is being built mainly for amd64 arch

File: README.md (hunk 5 above)
Comment (shiv-am0) [reply README.md:55]: The image will be pushed to the **harnesscommunity** docker account once all the manual testing and reviews are done.

File: README.md (hunk 2 above)
Comment (shiv-am0) [reply README.md]: I have updated the README already.

File: Dockerfile (hunk 1 above)
Comment (shiv-am0) [reply Dockerfile:2]: The plugin can be built and made to run for different arch with the help of the command: `docker buildx create --use --name harnesscommunity/drone-cobertura-s3 --platform linux/amd64,linux/arm64`

File: README.md (hunk 4 above)
Comment (shiv-am0) [reply README.md]: Fixed the descriptions in the README.

File: README.md (hunk 5 above)
Comment (shiv-am0) [reply README.md:55]: As per the discussion, custom plugins are pushed to harness-community/repo

File: main.go (hunk 6 above)
Comment (shiv-am0) [reply main.go]: Changed url to URL

File: main.go (hunk 7 above)
Comment (shiv-am0) [reply main.go]: Updated the descriptions

File: main.go (hunk 8 above)
Comment (shiv-am0) [reply main.go]: Description added

File: README.md (hunk 3 above)
Comment (shiv-am0) [reply README.md]: Directory name changed to "scripts"

**Review by Ompragash — APPROVED** [review]

**Pull request merged**